x/gov/client/cli/query.go

```diff
@@ -179,25 +180,34 @@ $ gaiacli query gov vote 1 cosmos1skjwj5whet0lpe65qaq4rpq03hjxlwd9nf39lk
 				return fmt.Errorf("Failed to fetch proposal-id %d: %s", proposalID, err)
 			}
 
-			// get voter address
 			voterAddr, err := sdk.AccAddressFromBech32(args[1])
 			if err != nil {
 				return err
 			}
 
-			// Construct query
 			params := gov.NewQueryVoteParams(proposalID, voterAddr)
 			bz, err := cdc.MarshalJSON(params)
 			if err != nil {
 				return err
 			}
 
-			// Query store
 			res, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/vote", queryRoute), bz)
 			if err != nil {
 				return err
 			}
 
+			var vote gov.Vote
+			// FIXME: We should check the error here but empty/non-existing votes will
```

Comment:
Can't we switch on the error type there and special case those?

Reply:
Error type where? This has to do with the way a vote option is JSON serialized. If there is no vote (it's been purged), then this will error and we won't get to the rest of the code.

Reply:
I think this is why the original author never bothered checking the error to begin with. Personally, I don't think we should error at all (cc @sunnya97)

Reply:
kk. Removed FIXME.